Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -566,17 +572,58 @@ void GivenOneProblemSavedAndArchived_WhenRestoreProblemById_ThenTheProblemShould
         assertFalse(actualProblem.isArchived());
     }
 
+    @DisplayName("Given two problems(A, B) saved and problem(A) is invisible" +
+            "When get invisible problems, " +
+            "Then should respond problem(A).")
+    @Test
+    void testGetInvisibleProblems() throws Exception {
+        int problemAId = 1, problemBId = 2;
+        saveProblems(problemAId, problemBId);
+        patchProblem(problemBId, patch -> patch.visible(true));
+
+        var expectInvisibleProblems = problemRepository.findAll().stream()
+                .filter(problem -> !problem.getVisible())
+                .filter(problem -> !problem.isArchived())
+                .map(ProblemItem::toProblemItem)
+                .collect(toList());
+        var actualInvisibleProblems = getInvisibleProblems(adminToken);
```

Comment:
Don't invoke `patchProblem` as you are not testing it.
I think you can do the same in the following way.

```java
saveProblems(problemAId);
var problemB = problemTemplate().id(problemBId).visible(false).build();
saveProblems(problemB);

var expectInvisibleProblems = [make problemB a problemItem];
...
```